README.md

```diff
@@ -194,6 +194,11 @@ Our project welcomes external contributions. Please refer to [CONTRIBUTING.md](C
 
 ## Change Log
 
+##### v0.*.*
+* Decrease memory footprint of the main data structures.
+* Added `max_size` option to limit the number of tracked clusters.
```

Comment:
I've added some documentation. I'll explain it also here to help with the conversation.

The main problem I want to solve is to make the algorithm stable when working with large log streams. If there are many templates, original version of Drain would save them all taking more and more memory. By introducing the cache we can set some limits for memory.

When it comes to the eviction policy, LRU is a natural fit because we want the model to quickly adapt to the changes in the template composition in the log stream. Other policies like removing clusters by the minimal count could work as well, but they run into the risk of slowly adapting. For example, if the cache was saturated with large clusters, model would completely loose its ability to adapt. LRU, on the other hand is well balanced. It will keep important templates, which appear from time to time and it will also adapt rapidly to the new templates.